sympy/matrices/common.py

```diff
@@ -2388,6 +2467,10 @@ def pow(self, exp, jordan=None):
             certain conditions, True specifies that jordan_pow should always
             be used if possible and False means it should not be used unless
             it is the only way to calculate the power.
+
+        cayley : bool, optional
+            If True then matrix exponention is done using cayley hamilton
+            theorem under certain conditions.
```

Comment:
You should mention the `certain condition` explicitly that the exponential should be an integer.